### DISCUSSION TITLE
no_std blockers

### DISCUSSION THREAD

**Discussion by amsam0** [yaahc/eyre#199] [Q&A]
Hi, as is stated in the README, eyre has not had no_std support for about 4 years now. However, after some very surface level source code browsing, it seems like the main reason no_std support was removed is because the `once_cell` crate was added as a dependency. `once_cell` does have no_std support though, with some caveats. Are there any larger reasons eyre cannot have no_std support (assuming a global allocator is available)?

**Comment by yaahc** [comment]
The main historical barrier has been that the `#[feature(error_in_core)]` is unstable. With it stabilizing in the next release I don't think there are any fundamental barriers, but I do believe we should hold off on adoption for a few compiler versions for MSRV reasons. I would not object to a PR adding back no_std support to the master branch for a future 1.0 release, and would also not object to a backport on the current major version once a few more releases have come out.

**Reply by amsam0** [reply]
Sounds good, I'll plan to send a PR once I need it for my project. Is increasing MSRV if the `std` feature is disabled an option?

**Reply by yaahc** [reply]
> Is increasing MSRV if the `std` feature is disabled an option?

By this, do you mean to make the MSRV conditional on a feature? I'm not sure if I'm misunderstanding what you mean or unfamiliar with this feature.

**Reply by amsam0** [reply]
Sorry for the confusion, I mean make it conditional. So we could say something like "if you have the `std` feature enabled, MSRV is X. If not, MSRV is X"

I didn't think about the Cargo.toml property, though. That probably wouldn't work then.
